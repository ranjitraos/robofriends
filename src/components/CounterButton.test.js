import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import CounterButton from './CounterButton';

describe('CounterButton tests', () => {
    it('renders CounterButton component correctly', () => {
        const mockColor = 'red'
        const counterButton = renderer
            .create(<CounterButton color={mockColor} />)
            .toJSON();
        expect(counterButton).toMatchSnapshot();
    });

    it('correctly increments the counter', async () => {
        const mockColor = 'red'
        const user = userEvent.setup()

        render(<CounterButton color={mockColor} />);
        const wrapper = screen.getByTestId('counter')

        await user.click(wrapper)
        expect(wrapper.innerHTML).toEqual('Count: 1')
    })
})