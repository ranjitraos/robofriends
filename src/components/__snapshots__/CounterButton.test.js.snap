// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton tests renders CounterButton component correctly 1`] = `
<button
  color="red"
  data-testid="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
